src/game_timer.cpp

```diff
@@ -332,16 +483,110 @@ void GameTimer::SplitSetDateDigits(int day, gt_tracker* gt_day, int mon, gt_trac
+		gt_year->digit_0.old_offset = year_offset_d0 - uv;
+		year_d0_flip = 1;
+		gt_year->digit_0.flip = 1;
+
+	}
+	else if (year_d0_flip == 1) {//middle of flipping
+		gt_year->digit_0.flip -= 0.1;
+		if (gt_year->digit_0.flip <= 0) {
+			year_d0_shown_offset = gt_year->digit_0.new_offset; //show new_offset once flipped past halfway
+		}
+		if (gt_year->digit_0.flip <= -1) {
+			gt_year->digit_0.flip = -1;
+			year_d0_flip = 0; //indicate done flipping;
+		}
+
+	}
+	else if (year_d0_flip == 0) {//done flipping, waiting for new flip
+		gt_year->digit_0.old_offset = year_offset_d0;
+	}
+
+	if (gt_year->digit_1.new_offset != year_offset_d1) { //need to flip
+		year_d1_shown_offset = gt_year->digit_1.old_offset;
+		gt_year->digit_1.new_offset = year_offset_d1;
+		gt_year->digit_1.old_offset = year_offset_d1 - uv;
+		year_d1_flip = 1;
+		gt_year->digit_1.flip = 1;
+
+	}
+	else if (year_d1_flip == 1) {//middle of flipping
+		gt_year->digit_1.flip -= 0.1;
+		if (gt_year->digit_1.flip <= 0) {
+			year_d1_shown_offset = gt_year->digit_1.new_offset; //show new_offset once flipped past halfway
+		}
+		if (gt_year->digit_1.flip <= -1) {
+			gt_year->digit_1.flip = -1;
+			year_d1_flip = 0; //indicate done flipping;
+		}
+
+	}
+	else if (year_d1_flip == 0) {//done flipping, waiting for new flip
+		gt_year->digit_1.old_offset = year_offset_d1;
+	}
+
+	if (gt_year->digit_2.new_offset != year_offset_d2) { //need to flip
+		year_d2_shown_offset = gt_year->digit_2.old_offset;
+		gt_year->digit_2.new_offset = year_offset_d2;
+		gt_year->digit_2.old_offset = year_offset_d2 - uv;
+		year_d2_flip = 1;
+		gt_year->digit_2.flip = 1;
+
+	}
+	else if (year_d2_flip == 1) {//middle of flipping
+		gt_year->digit_2.flip -= 0.1;
+		if (gt_year->digit_2.flip <= 0) {
+			year_d2_shown_offset = gt_year->digit_2.new_offset; //show new_offset once flipped past halfway
+		}
+		if (gt_year->digit_2.flip <= -1) {
+			gt_year->digit_2.flip = -1;
+			year_d2_flip = 0; //indicate done flipping;
+		}
+
+	}
+	else if (year_d2_flip == 0) {//done flipping, waiting for new flip
+		gt_year->digit_2.old_offset = year_offset_d2;
+	}
+
+	if (gt_year->digit_3.new_offset != year_offset_d3) { //need to flip
+		year_d3_shown_offset = gt_year->digit_3.old_offset;
+		gt_year->digit_3.new_offset = year_offset_d3;
+		gt_year->digit_3.old_offset = year_offset_d3 - uv;
+		year_d3_flip = 1;
+		gt_year->digit_3.flip = 1;
+
+	}
+	else if (year_d3_flip == 1) {//middle of flipping
+		gt_year->digit_3.flip -= 0.1;
+		if (gt_year->digit_3.flip <= 0) {
+			year_d3_shown_offset = gt_year->digit_3.new_offset; //show new_offset once flipped past halfway
+		}
+		if (gt_year->digit_3.flip <= -1) {
+			gt_year->digit_3.flip = -1;
+			year_d3_flip = 0; //indicate done flipping;
+		}
+
+	}
+	else if (year_d3_flip == 0) {//done flipping, waiting for new flip
+		gt_year->digit_3.old_offset = year_offset_d3;
+	}
 }
 
 void GameTimer::advance_time(float real_time_seconds_elapsed)
 {	
 	const int One_Day_Sec = 86400;
```

Comment:
Looking at the calendar, it is going real fast. Do we want Roy's time being a traffic cop to be by year or his whole life? If we want level progression, then it should be by year in my opinion

Reply:
We can change the time freely depending on what we decide to increment levels by